shards/run-test-ipfs.edn

```diff
@@ -0,0 +1,89 @@
+(deftrait Fragnova-Account :private-key "//Alice")
+(load-file "fragnova.edn")
+
+(def UPLOAD-CALL-DATA-EXCL-DATA-PARAM "0x0b00000000000000") ; encoded call data that defines a call to `protos.upload()` with certain hardcoded parameters. See: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fws.fragnova.network#/extrinsics/decode/0x0b0000000000000000
+
+(defwire upload-proto
+
+  .proto-data (ExpectBytes)
+
+  UPLOAD-CALL-DATA-EXCL-DATA-PARAM (HexToBytes) >= .call-data
+  .proto-data >> .seq .seq (Substrate.Encode [nil]) (AppendTo .call-data)
+
+  .call-data (Do (get Fragnova-Node :send-signed-extrinsic-as-http-request)) (Log "HTTP Response of protos.upload() is")
+
+  .proto-data (Hash.Blake2-256) (| (ToHex) (Log "Uploaded Proto With Proto Hash"))
+
+)
+
+
+(defwire convert-blake256-hash-to-ipfs-cid
+
+  .proto-hash (ExpectBytes) >= .proto-hash-duplicate
+
+  "0x0155a0e40220" (HexToBytes) (PrependTo .proto-hash-duplicate)
+  .proto-hash-duplicate (ToBase58) >= .ipfs-cid
+  "z" (PrependTo .ipfs-cid)
+  .ipfs-cid (Log "IPFS CID")
+
+)
+
+(defwire get-clamor-p2p-addresses
+
+  {"id" 1 "jsonrpc" "2.0" "method" "system_localListenAddresses" "params" []} (ToJson) (Http.Post RPC-SERVER) ; `system_localListenAddresses` return the string representation of the addresses we listen on, including the trailing /p2p/.
+  (FromJson) (ExpectTable) (Take "result")
+  (ExpectStringSeq) (Log "Clamor is listening to the following P2P addresses")
+)
+
+(defwire connect-to-clamor-p2p-addresses
+
+  .clamor-p2p-addresses (ExpectStringSeq)
+
+  ; Q: What is IPFS Swarm?
+  ; A: 'ipfs swarm' is a tool to manipulate the network swarm. The swarm is the component that opens, listens for, and maintains connections to other ipfs peers in the internet.
+
+  .clamor-p2p-addresses
+  (ForEach
+   (->
+    (Log "connecting to p2p address")
+    = .p2p-address
+    ["swarm" "peering" "add" .p2p-address] = .ipfs-args
+    "" (Process.Run :Executable "ipfs" :Arguments .ipfs-args)
+   )
+  )
+
+  (Maybe (-> "" (Process.Run :Executable "ipfs" :Arguments ["repo" "gc"]) nil)) ;  `ipfs repo gc` - Performs a garbage collection sweep on the IPFS Repo.
+
+)
+
+(defwire call-ipfs-block-get
+  .proto-hash-cid (ExpectString)
+
+  (Msg "Downloading IPFS block")
+
+  ["block" "get" .proto-hash-cid] = .ipfs-args
+  "" (Process.Run "ipfs" :Arguments .ipfs-args :Timeout 120) (StringToBytes) ; `ipfs block` - Interact with raw IPFS blocks. A block is identified by a Multihash passed with a valid CID.
+
+)
+
+(defwire main
+  (Setup
+   (get Fragnova-Node :setup)
+  )
+
+  "Proto-Indo-European" (ToBytes) = .proto-data
```

Comment:
@sinkingsugar @Kryptos-FR @guusw @tdbe @alebaffa @ShahFragcolor @ChelseaChX  - Giovanni, like I told you earlier, since the **current wire's variables are exposed to a child wire** whenever a child wire is called (i.e whenever `(Do <child wire name>)` is called) - **it makes it very difficult and cumbersome to use Shards "Input Output paradigm"** because a **shards scripter will always be worrying whether the variable name he initialized in the child wire already exists in the wire that calls the child wire!!!!!!!!**

For example, if you look at this example:

**PARENT WIRE:**
```clojure
"Proto-Indo-European" (ToBytes) (Do upload-proto)
**CHILD WIRE:**
```clojure
(defwire upload-proto
  (ExpectBytes) = **.proto-data**
  UPLOAD-CALL-DATA-EXCL-DATA-PARAM (HexToBytes) >= .call-data
  .proto-data >> .seq .seq (Substrate.Encode [nil]) (AppendTo .call-data)
  .call-data (Do (get Fragnova-Node :send-signed-extrinsic-as-http-request)) (Log "HTTP Response of protos.upload() is")
  .proto-data (Hash.Blake2-256) (| (ToHex) (Log "Uploaded Proto With Proto Hash"))
)
```

If the variable `.proto-data` in the CHILD WIRE (i.e in `upload-proto`) was initialized in the PARENT WIRE - then Shards will not even work - which is very frustrating!!!

**That's why I'm forced to write it like this in my code (which is extremely cumbersome) to "be safe":**
**PARENT WIRE:**
```clojure
  "Proto-Indo-European" (ToBytes) = .proto-data
  (Do upload-proto) ; `upload-proto` expects a variable called `proto-data`
```
**CHILD WIRE:**
```clojure
(defwire upload-proto
  **.proto-data** (ExpectBytes)
  UPLOAD-CALL-DATA-EXCL-DATA-PARAM (HexToBytes) >= .call-data
  .proto-data >> .seq .seq (Substrate.Encode [nil]) (AppendTo .call-data)
  .call-data (Do (get Fragnova-Node :send-signed-extrinsic-as-http-request)) (Log "HTTP Response of protos.upload() is")
  .proto-data (Hash.Blake2-256) (| (ToHex) (Log "Uploaded Proto With Proto Hash"))
)
```
But this approach is extremely cumbersome and doesn't utilize Shard's Input-Ouput Paradigm!

Reply:
Is this the reason why the IPFS tests fail with the new Substrate tag?